agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/classicsdk/TelemetryClientClassFileTransformer.java

```diff
@@ -405,6 +431,29 @@ private void overwriteTrackMetricMethod(MethodVisitor mv) {
       mv.visitEnd();
     }
 
+    private void overwriteIsDisabledMethod(MethodVisitor mv) {
+      mv.visitCode();
+      Label label0 = new Label();
+      mv.visitLabel(label0);
+      mv.visitVarInsn(ALOAD, 0);
+      mv.visitFieldInsn(
+          GETFIELD,
+          unshadedPrefix + "/TelemetryClient",
+          "configuration",
+          "L" + unshadedPrefix + "/TelemetryConfiguration;");
+      mv.visitMethodInsn(
+          INVOKEVIRTUAL,
+          unshadedPrefix + "/TelemetryConfiguration",
+          "isTrackingDisabled",
```

Comment:
I didn't find any tests for isTackingDisabled or isDisabled. How does customers use it? no ikey is only when they disable tracking, correct?

Reply:
they don't use it directly, the 2.x TelemetryClient calls `isDisabled()` is called in several places